src/lib/option.ts

```diff
@@ -76,118 +73,106 @@ export class Opt<T = unknown> extends Variable<T> {
-                } catch {
-                    // Do nuffin
-                }
-            }
-            currentCache[this._id] = newVal;
-            writeFile(cacheFile, JSON.stringify(currentCache, null, 2));
-        });
     }
 
     /**
-     * Initializes this option by attempting to read its default value from the default file.
-     * If found, sets the current value.
+     * Set the given configuration value and write it to disk, if specified.
+     *
+     * @param value - The new value.
+     * @param writeDisk - Whether to write the changes to disk. Defaults to true.
      */
-    public createDefault(): void {
-        const rawData = readFile(defaultFile);
-
-        let defaultData: Record<string, unknown> = {};
-
-        if (rawData && rawData.trim() !== '') {
-            try {
-                defaultData = JSON.parse(rawData) as Record<string, unknown>;
-            } catch {
-                // do nuffin
-            }
+    public set(value: T, { writeDisk = true }: WriteDiskProps = {}): void {
+        if (value == this.get()) {
+            // If nothing actually changed, exit quick
+            return;
         }
 
-        const defaultVal = defaultData[this._id];
+        super.set(value);
 
-        if (defaultVal !== undefined) {
-            this.set(defaultVal as T);
+        if (writeDisk) {
+            const raw = readFile(CONFIG);
+            let config: Record<string, unknown> = {};
+            if (raw && raw.trim() !== '') {
+                try {
+                    config = JSON.parse(raw) as Record<string, unknown>;
+                } catch (error) {
+                    // Last thing we want is to reset someones entire config
+                    // so notify them instead
+                    console.error(`Failed to load config file: ${error}`);
+                    Notify({
+                        summary: 'Failed to load config file',
+                        body: `${error}`,
+                        iconName: icons.ui.warning,
+                    });
+
+                    errorHandler(error);
+                }
+            }
+            config[this._id] = value;
+            writeFile(CONFIG, JSON.stringify(config, null, 2));
         }
     }
 
     /**
      * Resets the value of this option to its initial value if not persistent and if it differs from the current value.
      *
+     * @param writeDisk - Whether to write the changes to disk. Defaults to true.
      * @returns Returns the option's ID if reset occurred, otherwise undefined.
      */
-    public reset(): string | undefined {
+    public reset(writeDiskProps: WriteDiskProps = {}): string | undefined {
         if (this.persistent) {
             return undefined;
         }
 
         const current = this.get();
 
-        if (JSON.stringify(current) !== JSON.stringify(this.initial)) {
-            this.set(this.initial);
+        if (current !== this.initial) {
```

Comment:
```suggestion
        let currentValue: string | T = this.get();
        currentValue = typeof this.get() === 'object' ? JSON.stringify(currentValue) : currentValue;

        let initialValue: string | T = this.get();
        initialValue = typeof this.get() === 'object' ? JSON.stringify(initialValue) : initialValue;

        if (currentValue !== initialValue) {
```
This condition will always evaluate to false for objects. Let's stringify if the value is an object and then compare. Otherwise just compare the value as is if not object.